DataFormats/Luminosity/interface/PCC.h

```diff
@@ -0,0 +1,74 @@
+#ifndef PCC_h
+#define PCC_h
+/** \class reco::PCC
+ *  
+ * Reconstructed PCC object that will contain the moduleID, BX, and counts.
+ *
+ * \authors: Sam Higginbotham shiggib@cern.ch and Chris Palmer capalmer@cern.ch
+ * 
+ *
+ */
+#include <algorithm>
+#include <string>
+#include <sstream>
+#include <iostream>
+#include <vector>
+
+namespace reco {
+class PCC {
+static constexpr unsigned int nEmBX=3564;//Empty BX to fill with counts
+    public:
+        PCC() : m_events(nEmBX){}
+        ////////////////////////////////////////////
+        void Increment(int mD,int BXid,int count){
+            std::vector<int>::iterator it;
+            it = std::find(m_ModID.begin(), m_ModID.end(), mD);
+            size_t modIndex = it - m_ModID.begin();
+
+            if(it == m_ModID.end()){
+                std::vector<int> m_empBX(nEmBX, 0);
+                m_ModID.push_back(mD);
+                m_counts.push_back(m_empBX); 
+            }
+           m_counts[modIndex][BXid] += count; 
+        }
+        ////////////////////////////////////////////
+        void eventCounter(int BXid){
+            m_events[BXid]++;
+        }
+        ////////////////////////////////////////////
+        std::vector<std::vector<int>> const & read_counts() const {
+                return(m_counts);
+            }
+        ////////////////////////////////////////////
+        void printVector()
+        { 
+            int irow = 0;
+            std::vector< std::vector<int> >::const_iterator row; 
+            std::vector<int>::const_iterator col; 
+
+            for (row = m_counts.begin(); row != m_counts.end(); ++row)
+            { 
+            
+               std::cout << m_ModID[irow] << " :";
+               for (col = row->begin(); col != row->end(); ++col)
+               { 
+                  std::cout << *col << " "; 
+               } 
+               irow++;
+               std::cout<<std::endl;
+            } 
+
+        }
+
+      private:
+        std::vector<std::vector<int> > m_counts;
```

Comment:
We need pixel cluster counts with module ID x BCID granularity.  We like this scheme because we do not need to know the module IDs beforehand.  I think the only way to flatten would be to hardcode all the module IDs beforehand (which isn't a nice option).

Do you have an alternative suggestion?